src/app/icd/BUILD.gn

```diff
@@ -16,11 +16,20 @@ import("//build_overrides/chip.gni")
 import("icd.gni")
 
 # ICD Server sources and configurations
+config("notifier-config") {
+  defines =
+      [ "ICD_MAX_NOTIFICATION_SUBSCRIBERS=${icd_max_notification_subscribers}" ]
```

Comment:
This adds a command-line define for compilers. This is inconvenient when integrating into 3rd party build systems that are not gn-based. It would be prefferable to use a buildconfig_header.

Reply:
moved it to the `app_buildconfig` buildconfig_header